examples/documentation_examples/examples.go

```diff
@@ -2956,3 +2957,172 @@ func StableAPIExamples() {
+	if err != nil {
+		return err
+	}
+	defer sess.EndSession(ctx)
+
+	var adoptablePetsCount int32
+	err = mongo.WithSession(ctx, sess, func(ctx mongo.SessionContext) error {
+		// Count the adoptable cats
+		adoptableCatsOutput := "adoptableCatsCount"
+		cursor, err := db.Collection("cats").Aggregate(ctx, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableCatsOutput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(ctx)
+		resp := cursor.Current.Lookup(adoptableCatsOutput)
+		adoptableCatsCount, ok := resp.Int32OK()
+		if !ok {
+			return fmt.Errorf("failed to find int32 field %q in document %v", adoptableCatsOutput, cursor.Current)
+		}
+		adoptablePetsCount += adoptableCatsCount
+
+		// Count the adoptable dogs
+		adoptableDogsOutput := "adoptableDogsCount"
+		cursor, err = db.Collection("dogs").Aggregate(ctx, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableDogsOutput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(ctx)
+		resp = cursor.Current.Lookup(adoptableDogsOutput)
+		adoptableDogsCount, ok := resp.Int32OK()
+		if !ok {
+			return fmt.Errorf("failed to find int32 field %q in document %v", adoptableDogsOutput, cursor.Current)
+		}
+		adoptablePetsCount += adoptableDogsCount
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	// End Snapshot Query Example 1
+	require.Equal(mt.T, int32(3), adoptablePetsCount, "expected adoptablePetsCount to be 3, got %v", adoptablePetsCount)
+	return nil
+}
+
+func snapshotQueryRetailExample(mt *mtest.T) error {
+	client := mt.Client
+	db := mt.DB
+
+	// Start Snapshot Query Example 2
+	ctx := context.TODO()
+
+	sess, err := client.StartSession(new(options.SessionOptions).SetSnapshot(true))
+	if err != nil {
+		return err
+	}
+	defer sess.EndSession(ctx)
+
+	var totalDailySales int32
+	err = mongo.WithSession(ctx, sess, func(ctx mongo.SessionContext) error {
+		// Count the total daily sales
+		totalDailySalesOutput := "totalDailySales"
+		cursor, err := db.Collection("sales").Aggregate(ctx, mongo.Pipeline{
+			bson.D{{"$match",
+				bson.D{{"$expr",
+					bson.D{{"$gt",
+						bson.A{"$saleDate",
+							bson.D{{"$dateSubtract",
+								bson.D{
+									{"startDate", "$$NOW"},
+									{"unit", "day"},
+									{"amount", 1},
+								},
+							}},
+						},
+					}},
+				}},
+			}},
+			bson.D{{"$count", totalDailySalesOutput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(ctx)
+		resp := cursor.Current.Lookup(totalDailySalesOutput)
+
+		var ok bool
+		totalDailySales, ok = resp.Int32OK()
+		if !ok {
+			return fmt.Errorf("failed to find int32 field %q in document %v", totalDailySalesOutput, cursor.Current)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	// End Snapshot Query Example 2
+	require.Equal(mt.T, int32(1), totalDailySales, "expected total to be 1, got %v", totalDailySales)
```

Comment:
Optional: Same as above, `require.Equal` prints rich assertion failure messages, so you can simplify this line.

E.g.
```go
require.Equal(mt, int32(1), totalDailySales, "expected 1 total daily sale")
```